text/0225-configuration.md

```diff
@@ -0,0 +1,340 @@
+
+### Environment Variable Interface Limitations
+
+The environment variable interface suffers from the following identified limitations:
+
+* **Flat**. Structured data is only allowed by using higher-level naming or data encoding schemes. Examples of configuration limited by lack of structured configuration include:
+  * Configuring multiple span processors, periodic metric readers, or log record processors.
+  * Configuring views.
+  * Configuring arguments for parent based sampler (sampler parent is remote and sampled vs. not sampled, sampler when parent is local and sampled vs. not sampled).
+* **Runtime dependent**. Different systems expose this interface differently (Linux, BSD, Windows). This usually means unique instructions are required to properly interact with the configuration interface on different systems.
+* **Limited values**. Many systems only allow string values to be used, but OpenTelemetry specifies many configuration values other than this type. For example, OTEL_RESOURCE_ATTRIBUTES specifies a list of key value pairs to be used as resource attributes, but there is no way to specify array values, or indicate that the value should be interpreted as non-string type.
+* **Limited validation**. Validation can only be performed by the receiver, there is no meta-configuration language to validate input.
+* **Difficult to extend**. It’s difficult to anticipate the requirements of configuring custom extension components (processors, exporters, samplers, etc), and likely not practical to represent them in a flat structure. As a result, the environment variable interface is limited to components explicitly mentioned in the specification.
+* **Lacks versioning**. The lack of versioning support for environment variables prevents evolution over time.
+
+## Explanation
+
+Using a configuration model or configuration file, users can configure all options currently available via environment variables.
+
+### Goals
+
+* The configuration must be language implementation agnostic. It must not contain structure or statements that only can be interpreted in a subset of languages. This does not preclude the possibility that the configuration can have specific extensions included for a subset of languages, but it does mean that the configuration must be interpretable by all implementation languages.
+* Broadly supported format. Ideally, the information encoded in the file can be decoded using native tools for all OpenTelemetry implementation languages. However, it must be possible for languages that do not natively support an encoding format to write their own parsers.
+* The configuration format must support structured data. At the minimum arrays and associative arrays.
+* The format must support at least null, boolean, string, double precision floating point (IEEE 754-1985), and signed 64 bit integer value types.
```

Comment:
Should `null` be not simply "non-existing"? Meaning that some things are "required" and some are "optional". I do not think we need to have `null` in the configuration file. Therfore, I do not see why the format must support `null`.

Reply:
```suggestion
* The format must support at least boolean, string, double precision floating point (IEEE 754-1985), and signed 64 bit integer value types.
```

Reply:
Thanks @pellared!